main.c

```diff
@@ -354,9 +363,26 @@ int main(void)
         switch (USB_getConnectionState())
         {
             case ST_ENUM_ACTIVE:
-                if(uartmuxState != UARTMUXSEL_MODE_MSP430)
+                uartmuxState |= UARTMUXSEL_MODE_LOCK_MSP430;
+                if((uartmuxState & 0x0F) != UARTMUXSEL_MODE_MSP430) {
+                    /*
+                     * request UART mux switch by informing CC1352.
+                     * CC1352 is configured to use the BOOT pin as 
+                     * interrupt for the I2C slave, and will read 
+                     * the I2C slave status and further queries
+                     * uartmuxState.
+                     */
+                    i2cSlaveState =  BCF_I2CSLAVE_UARTMUX_CHANGE;                    
+                    if(uartmux_lock_trigger) {
+                        hal_ext_boot(TRUE);
+                        __delay_cycles(2000);
```

Comment:
Is this a magic number? What duration is required for the CC1352 to recognize the interrupt?

Reply:
Will update in the next commit, reused the same values used for the bootloader trigger delay, delay to recognize the interrupt should be less than this.